### DISCUSSION TITLE
Charging_Amps

### DISCUSSION THREAD

**Discussion by israndy** [tdorssers/TeslaPy#90] [General]
You have a note under the Commands section of the TeslaPy readme that I hope you can update:

>  requires car version 2021.36 or higher. CHARGING_AMPS can be set to less than 5, by calling the API twice.

I have been working with this for my App TesSense and noted thru experience but have not seen anywhere on the net that if you try to set the amps to 2 it will set them to 3, and if you try to set them to 1 or less it will set them to 2

So 4 amps = 4 amps, 3=3, 2=3, 1=2, 0=2, etc.

I had hoped in my solar Tesla charging app that I could actually set the charger to 0 amps instead of turning it off, for times like around sunset so the relay wouldn't keep flipping on and off as the solar power waned, but I will have to come up with another strategy, or live with the 2 amps charge rate until the sun is officially down

**Comment by Mustang5705** [comment]
In fact, it is useless to go below 4A because the base consumption of the vehicle during charging is too close and the efficiency of the real charge is low with a lower amperage. Anyway, great app. Respect to the developers.

**Comment by israndy** [comment]
Which is probably why Tesla set it up the way they did, but my point was that the documentation is wrong on TeslaPy and hoped I helped clarify

**Comment by purcell-lab** [comment]
I find I get different results with different connectors.

With the UMC, if I set charging_amps to 0 it actually stops charging and drawing power from the socket.
With the HWPC and TWC, if I set charging_amps to 0 it only goes down to 1 or 2 amps and continues to draw power.

I have three phase power on my TWC (pity the API says 2 phases), but at 3A x3 phases = 9A I'm getting the same as a UMC, so don't write off all the values below 5A.  I also find it really useful to match against solar having the lower ranges.  If I was limited to 5 A x 3 Phase x 240V = 3,600 W, which is hard to have excess for a standard 5 kW solar inverter.  I would love to see some API results when using _charge on solar_, but that isn't released to my market :-(

UMC:
![image](https://github.com/tdorssers/TeslaPy/assets/79175134/7aecb73d-cf90-4ca8-9b7d-3c22b8aedce5)
![image](https://github.com/tdorssers/TeslaPy/assets/79175134/98d4d106-743e-49d8-9987-75ee15598e89)
![image](https://github.com/tdorssers/TeslaPy/assets/79175134/68b3c913-fb0f-425d-b8b6-2ee9e33d1928)
![image](https://github.com/tdorssers/TeslaPy/assets/79175134/814ced7b-efb4-4cee-960a-f58790b4635d)
![image](https://github.com/tdorssers/TeslaPy/assets/79175134/b0c9f53d-7039-4df7-8549-458305019372)

TWC (gen 3):
![image](https://github.com/tdorssers/TeslaPy/assets/79175134/0794231f-0453-4a68-8f42-7d48a27c843d)
![image](https://github.com/tdorssers/TeslaPy/assets/79175134/c78e5f9d-7a1e-4760-ac87-7f5f2060d0e3)
![image](https://github.com/tdorssers/TeslaPy/assets/79175134/e831a8c1-80c4-4058-b10d-4b6388c1b0e1)
![image](https://github.com/tdorssers/TeslaPy/assets/79175134/183cb88e-1222-4caa-acf8-8e466e16e9f1)

**Reply by israndy** [reply]
Interesting, which UMC are you using? I have the portable charger that came with the original Model S and I get different results from the ones you show, 0 means 0 for mine, then 1 = 2 actual, 2=3, 3=3, 4=4 and so on

**Reply by purcell-lab** [reply]
I have the Gen2 UMC that came with a 2020 model 3, it is capable of upto 32A single phase.

I understand the Gen1 UMC came with earlier model 3 and could support upto 40A.
